src/ecommerce-common/01_network.tf

```diff
@@ -14,4 +14,22 @@ resource "azurerm_private_dns_a_record" "ingress" {
   resource_group_name = local.internal_dns_zone_resource_group_name
   ttl                 = 3600
   records             = [var.ingress_load_balancer_ip]
+}
+
+data "azurerm_subnet" "aks_subnet" {
+  name                 = local.aks_subnet_name
+  virtual_network_name = local.vnet_name
+  resource_group_name  = local.vnet_resource_group_name
+}
+
+data "azurerm_private_dns_zone" "cosmos" {
+  name                = local.cosmos_dns_zone_name
+  resource_group_name = local.cosmos_dns_zone_resource_group_name
+}
+
+resource "azurerm_private_dns_zone" "privatelink_ecommerce_redis" {
+  name                = "privatelink.ecommerce.redis.azure.com"
+  resource_group_name = local.vnet_resource_group_name
+
+  tags = var.tags
 }
```

Comment:
```suggestion
```
U can use only one private dns zone for resource type and subscription. So u need to reuse privatelink.redis.cache.windows.net